### PR TITLE
Added huggingface hub login

### PR DESCRIPTION
**Pull request by Walchshofer**

### CODE REVIEW

**Comment by Walchshofer** [comment]
added huggingface hub login

File: app/settings.py [hunk 1]
```diff
@@ -60,12 +60,14 @@ def __init__(self):
         if self.root_path.endswith("app"):
             self.root_path = self.root_path[:-4]
 
-        self.model_root_path = os.path.join(self.root_path, "models")
+        #self.model_root_path = os.path.join(self.root_path, "models")
+        self.model_root_path = "/home/pwalch/textgen/models" # changed the model path
```
Comment (iGavroche) [inline app/settings.py:64]: do not make this change for all users please.

File: app/settings.py [hunk 2]
```diff
@@ -60,12 +60,14 @@ def __init__(self):
...
         self.backend_root_path = os.path.join(self.root_path, "app")
 
         # let's ensure the models and backend path is accurate when mounting docker volumes
         if self.root_path.startswith("/code"):
-            self.model_root_path = "/models"
+            #self.model_root_path = "/models"
+            self.model_root_path = "/home/pwalch/textgen/models" # changed the model path
```
Comment (iGavroche) [inline app/settings.py:70]: this changed should be managed through the `.env` file.
Also note that "/models" refers to "<your project's root folder>/models".

File: bootstrap_models.py [hunk 3]
```diff
@@ -1,11 +1,21 @@
...
 DEFAULT_MODEL = "openlm-research/open_llama_3b"
 MODEL_DIR = "./models"
 
+def login_to_huggingface_hub():
+    load_dotenv()
+    hf_token = os.getenv("HUGGINGFACE_TOKEN")
+
+    # Login to Hugging Face Hub
+    hf_api = HfApi()
+    user = hf_api.whoami(hf_token)
```
Comment (iGavroche) [inline bootstrap_models.py:17]: make this conditional to there being a huggingface token present in `.env`. If not, continue with unauthenticated session.

**Review by iGavroche — CHANGES_REQUESTED** [review]
Thanks for submitting a PR. Please review the comments and also join our Discord (link in Readme.md I believe) if interested in developing further.

**Pull request closed without merging**